src/Symfony/Component/Security/Core/Authorization/Voter/Vote.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Core\Authorization\Voter;
+
+/**
+ * A Vote is returned by a Voter and contains the access (granted, abstain or denied). It can also contains a reason
+ * explaining the why of the access which has been decided.
+ *
+ * @author Dany Maillard <danymaillard93b@gmail.com>
+ */
+final class Vote
+{
+    use AccessTrait;
+
+    private $reason;
+    private $parameters;
+
+    /**
+     * @param int $access One of the VoterInterface::ACCESS_* constants
+     */
+    private function __construct(int $access, string $reason = '', array $parameters = [])
+    {
+        $this->access = $access;
+        $this->reason = $reason;
+        $this->parameters = $parameters;
+    }
+
+    public static function create(int $access, string $reason = '', array $parameters = []): self
+    {
+        return new self($access, $reason, $parameters);
+    }
+
+    public static function createGranted(string $reason, array $parameters = []): self
+    {
+        return new self(VoterInterface::ACCESS_GRANTED, $reason, $parameters);
+    }
+
+    public static function createAbstrain(string $reason, array $parameters = []): self
+    {
+        return new self(VoterInterface::ACCESS_ABSTAIN, $reason, $parameters);
+    }
+
+    public static function createDenied(string $reason, array $parameters = []): self
+    {
+        return new self(VoterInterface::ACCESS_DENIED, $reason, $parameters);
+    }
+
+    public function merge(self $vote): void
```

Comment:
I don't really see the value of this method. It doesn't fullfill what I would call "merge" (i.e. it only merges the reason, what happens if we pass denied into the merge of an granted vote?) and it's used only in one specific case. I would rather add a `setReason()` method and doing this directly in the voter.